README.md

```diff
@@ -5,8 +5,148 @@ This Cabal Git repository contains the following packages:
  * [Cabal](Cabal/README.md): the Cabal library package ([license](Cabal/LICENSE))
  * [cabal-install](cabal-install/README.md): the package containing the `cabal` tool ([license](cabal-install/LICENSE))
 
-See [HACKING.md](HACKING.md) for information about contributing and building
-from git cloned sources.
-
 The canonical upstream repository is located at
 https://github.com/haskell/cabal.
+
+Installing Cabal
+----------------
+
+Assuming that you have a pre-existing, older version of `cabal-install`,
+run:
+
+~~~~
+cabal install cabal-install
+~~~~
+
+To get the latest version of `cabal-install`. (You may want to `cabal
+update` first.)
+
+To install the latest version from the Git repository, clone the
+Git repository and then run:
+
+~~~~
+(cd Cabal; cabal install)
+(cd cabal-install; cabal install)
+~~~~
+
+Building Cabal for hacking
+--------------------------
+
+The current recommended way of developing Cabal is to use the
+`new-build` feature which shipped in cabal-install-1.24.  Assuming
```

Comment:
I hadn't noticed, but I don't think it changes anything per se about using `setup-dev.sh`... *until* I can try out new-build and observe that it works for me :).

I've been using `ghcid` to do continuous recompilation, not sure what the status of `ghcid` + new-build is. Once I see that working, I'll be happy to delete `setup-dev.sh` :).